### PR TITLE
DOC: Extension whatsenw

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes https://github.com/pandas-dev/pandas/issues/20532

### CODE REVIEW

**Comment by TomAugspurger** [comment]
Whoops, looks like I branched off the wrong branch. Oh well, fixing some other warnings too I guess :)

File: doc/source/whatsnew/v0.23.0.txt [hunk 1]
```diff
@@ -299,6 +299,61 @@ Supplying a ``CategoricalDtype`` will make the categories in each column consist
     df['A'].dtype
     df['B'].dtype
 
+Extension Types
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.23.0.txt]: Maybe a more descriptive title like "Extending pandas with custom data types" ?

File: doc/source/whatsnew/v0.23.0.txt [hunk 2]
```diff
@@ -299,6 +299,61 @@ Supplying a ``CategoricalDtype`` will make the categories in each column consist
...
+arrays as columns in a DataFrame or values in a Series. This allows third-party
+libraries to implement extensions to NumPy's types, similar to how pandas
+implemented categoricals, datetimes with timezones, periods, and intervals.
+
+As a demonstration, we'll use cyberpandas_, which provides an ``IPArray`` type
+for storing ip addresses.
+
+.. code-block:: ipython
+
+   In [1]: import pandas as pd; from cyberpandas import IPArray
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.23.0.txt]: two lines?

File: doc/source/whatsnew/v0.23.0.txt [hunk 3]
```diff
@@ -299,6 +299,61 @@ Supplying a ``CategoricalDtype`` will make the categories in each column consist
...
+   In [2]: values = IPArray([
+      ...:     0,
+      ...:     3232235777,
+      ...:     42540766452641154071740215577757643572
+      ...: ])
+      ...:
+      ...:
+
+``IPArray`` isn't a normal 1-D NumPy array, but because it's a pandas
+ExtensionArray, it can be stored properly inside pandas' containers.
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.23.0.txt]: quote ExtensionArray ?

**Review by jorisvandenbossche — COMMENTED** [review]
Looks good, some minor comments

**Pull request merged**